doc/getting-started/apl-nuc.rst

```diff
@@ -133,6 +140,165 @@ complete this setup.
+      Check ACRN efi boot event
+      Clean all ACRN efi boot event
+      Check linux bootloader event
+      Clean all Linux bootloader event
+      Add new ACRN efi boot event
+      Create loader.conf
+      Add default (5 seconds) boot wait time
+      Add default boot to ACRN
+      Getting latest Service OS kernel version: kernel-org.clearlinux.iot-lts2018-sos.4.19.34-45
+      Getting current Service OS kernel version: kernel-org.clearlinux.iot-lts2018-sos.4.19.13-1901141830
+      Replacing root partition uuid in acrn.conf
+      Replace with new SOS kernel in acrn.conf
+      Service OS setup done!
+      Rebooting Service OS to take effects.
+      Rebooting.
+
+   .. note::
+      This script is using ``/dev/sda1`` as default EFI System Partition (ESP). The ESP
+      may be different based on your hardware and then you should specify it directly with ``-e`` option.
+      Here is an example for setup SOS on NVMe SSD: ``sudo sh acrn_quick_setup.sh -s 28960 -e /dev/nvme0n1p1``
+
+   .. note::
+      If you don't need reboot automatically after set up SOS, then you should run this command:
+      ``sudo sh acrn_quick_setup.sh -s 28960 -d``
+
+#. After the system reboots and login as the clear user, you may need to check the ``dmesg`` to make sure
+   the SOS is boot successfully.
+
+   .. code-block:: console
+
+      $ dmesg | grep ACRN
+      [    0.000000] Hypervisor detected: ACRN
+      [    1.220887] ACRNTrace: Initialized acrn trace module with 4 cpu
+      [    1.224401] ACRN HVLog: Initialized hvlog module with 4 cpu
+
+#. If you want to continue to set up a Guest OS after boot SOS, then you can run
+   ``sudo sh acrn_quick_setup.sh -u 28960`` to get your UOS ready.
+
+   .. code-block:: console
+
+      $ sudo sh acrn_quick_setup.sh -u 28960
+      Password:
+      Upgrading UOS...
+      Downloading UOS image: https://download.clearlinux.org/releases/28960/clear/clear-28960-kvm.img.xz
+        % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                       Dload  Upload   Total   Spent    Left  Speed
+       14  248M   14 35.4M    0     0   851k      0  0:04:57  0:00:42  0:04:15  293k
+
+   After download is completed, you'll get this output.
+
+   .. code-block:: console
+
+      Unxz UOS image: clear-28960-kvm.img.xz
+      Get UOS image: clear-28960-kvm.img
+      Upgrade UOS done...
+      Now you can run this command to start UOS...
+      $ sudo /root/launch_uos_28960.sh
+
+   .. note::
+      If you have a local UOS image which is named ``clear-28960-kvm.img.xz`` or it's just uncompressed into
+      ``/root`` folder which is named ``clear-28960-kvm.img``, then you can run
+      ``sudo sh acrn_quick_setup.sh -u 28960 -k`` to skip downloading it again and set up UOS directly.
+
+#. Now you can run ``sudo /root/launch_uos_28960.sh`` to launch UOS.
+
+   .. code-block:: console
+
+      $ sudo /root/launch_uos_28960.sh
```

Comment:
I'm getting a failure when I run the launch_uos script:
```
clear@clr-sos-guest ~ $ sudo /root/launch_uos_29100.sh
cpu1 online=1
cpu2 online=1
cpu3 online=1
passed gvt-g optargs low_gm 64, high_gm 448, fence 8
SW_LOAD: get kernel path /usr/lib/kernel/default-iot-lts2018
SW_LOAD: get bootargs root=/dev/vda3 rw rootwait maxcpus=1 nohpet console=tty0 console=hvc0   console=ttyS0 no_timer_check ignore_loglevel log_buf_len=16M   consoleblank=0 tsc=reliable i915.avail_planes_per_pipe=0x070F00   i915.enable_hangcheck=0 i915.nuclear_pageflip=1 i915.enable_guc_loading=0   i915.enable_guc_submission=0 i915.enable_guc=0
VHM api version 1.0
open hugetlbfs file /run/hugepage/acrn/huge_lv1/D279543825D611E8864ECB7A18B34643
open hugetlbfs file /run/hugepage/acrn/huge_lv2/D279543825D611E8864ECB7A18B34643
level 0 free/need pages:0/0 page size:0x200000
level 1 free/need pages:0/2 page size:0x40000000
to reserve more free pages:
to reserve pages (+orig 0): echo 2 > /sys/kernel/mm/hugepages/hugepages-1048576kB/nr_hugepages
now enough free pages are reserved!

try to setup hugepage with:
    level 0 - lowmem 0x0, biosmem 0x0, highmem 0x0
    level 1 - lowmem 0x80000000, biosmem 0x0, highmem 0x0
total_size 0x180000000

mmap ptr 0x0x7f48e10e3000 -> baseaddr 0x0x7f4900000000
mmap 0x80000000@0x7f4900000000
touch 2 pages with pagesz 0x40000000

really setup hugepage with:
    level 0 - lowmem 0x0, biosmem 0x0, highmem 0x0
    level 1 - lowmem 0x80000000, biosmem 0x0, highmem 0x0
polling 33...
Listening 33...
acrn-dm: Could not open backing file: clear-29100-kvm.img: No such file or directory
                                                                                    Could not open backing file: No such file or directory
       mngr_client_new: Failed to accept from fd 33, err: Invalid argument
                                                                          Stop listening 33...
                                                                                              Stop polling 33...
                                                                                                                Unable to init vdev (2)
clear@clr-sos-guest ~ $ 

```

Reply:
I'll check this on my board, and could you please provide these outputs:
``cat /root/launch_uos_29100.sh | grep clear-29100-kvm.img``
``ls -l /root/clear-29100-kvm.img``

Reply:
UOS launch fail is due to ``$uos_img`` isn't getting the full directory. It's fixed in latest release of script.

Reply:
My NUC is in the office so I'll need to get you that information tomorrow (I'll fetch a new script and try in the morning too).